From be8cb61c23eacb91a26ae6fa3305f92d7d79a51f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pasi=20Sj=C3=B6holm?= <pasi.sjoholm@jollamobile.com>
Date: Mon, 2 Mar 2015 10:09:35 +0200
Subject: [PATCH 15/22] dhcpv6: Rate limit requests when no address is received

When reply does not contain addresses and has not set error in the
advertise phase, it is possible to end up with a loop of requests
(eg. when IA_TA requested from Cisco EPC3940L).

(cherry picked from commit 064385639bcb519ab86c8b49b172e406d81a3702)
---
 src/dhcpv6.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dhcpv6.c b/src/dhcpv6.c
index d3ff127..bdb3b98 100644
--- a/src/dhcpv6.c
+++ b/src/dhcpv6.c
@@ -1107,7 +1107,7 @@ static void re_cb(enum request_type req_type, GDHCPClient *dhcp_client,
 			if (!option) {
 				switch (req_type) {
 				case REQ_REQUEST:
-					dhcpv6_request(dhcp, true);
+					do_resend_request(dhcp);
 					break;
 				case REQ_REBIND:
 					dhcpv6_rebind(dhcp);
-- 
1.9.1

